acmwebsite/controllers/survey.py

```diff
@@ -1,34 +1,68 @@
-from tg import expose, redirect, validate, flash, url, lurl, abort, require, request
-from tg.predicates import has_permission, not_anonymous
+from tg import abort, expose, flash, redirect, request, require
+from tg.predicates import has_permission
 
 from acmwebsite.lib.base import BaseController
-from acmwebsite.lib.helpers import log
-from acmwebsite.model import DBSession, Survey, SurveyResponse, SurveyData, User
+from acmwebsite.model import DBSession, Survey, SurveyData, SurveyResponse
 
-def survey_fields(survey):
-    return [{'name': f.name, 'type': f.type} for f in survey.fields]
-
-def response_to_dict(response):
-    out = {'name': response.name, 'email': response.email}
-    for item in response.data:
-        out[item.field.name] = item.field.type_object().from_contents(item.contents)
-    return out
 
 class SurveyController(BaseController):
     def __init__(self, survey):
         self.survey = survey
 
+    @expose('acmwebsite.templates.survey_results')
+    @require(has_permission('admin'))
+    def results(self, number=None, order_by=None, reverse=False):
+        if type(reverse) is str:
+            reverse = reverse == 'True'
+
+        responses = self._survey_responses()
+        if order_by:
+            responses.sort(key=lambda x: x.get(order_by), reverse=reverse)
+
+        survey_title = (self.survey.title or
+                        (self.survey.meeting and self.survey.meeting.title) or
+                        'Survey')
+        return {
+            'survey': self.survey,
+            'title': survey_title,
+            'count': len(responses),
+            'responses': responses,
+            'fields': self.survey.field_metadata(),
+            'order_by': order_by,
+            'reverse': reverse,
+        }
+
     @expose('json')
     @require(has_permission('admin'))
-    def results(self, number=None):
-        responses = self.survey.responses or []
-        responses = [response_to_dict(r) for r in responses]
+    def results_json(self, number=None):
```

Comment:
Combine into the results method. There is no reason that these two endpoints need separate methods in the controller.